src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs

```diff
@@ -70,6 +71,9 @@ internal class CrossValSummaryRunner<TMetrics> : IRunner<RunDetail<TMetrics>>
 
             // Get the model from the best fold
             var bestFoldIndex = BestResultUtil.GetIndexOfBestScore(trainResults.Select(r => r.score), _optimizingMetricInfo.IsMaximizing);
+            // bestFoldIndex will be -1 if the optimization metric for all folds is NaN.
+            // In this case, return model from the first fold.
+            bestFoldIndex = bestFoldIndex != -1 ? bestFoldIndex : 0;
```

Comment:
I'd also look into places where the metric gets compared. 

For instance below in `GetIndexClosestToAverage()`:
https://github.com/dotnet/machinelearning/blob/9726b4103e066d410a2b0728fb3e2058c081e15c/src/Microsoft.ML.AutoML/Experiment/Runners/CrossValSummaryRunner.cs#L86-L100

While `GetIndexClosestToAverage()` could be adjusted to handle CV folds returning `NaN`, it would be better to remove the function and instead return a new instance of the metric class w/ the actual averages. The current function was created before the AutoML code had access to create a new instance of the metric classes, so it just returned the closest to the average of the folds.

I would also look at a bit further to where AutoML compares the return metrics from each model in the sweep, where it chooses which is the best. The comparison to `NaN` may also be there.

Reply:
I have changed the calculation of average to exclude any folds with NaN metrics. In finding the best run, there are not any comparisons to NaN, but an index of -1 is returned if *all* the runs have NaN metric. I have added a warning there checking for that case.

As for returning a Metrics object containing the averages of the metrics across the folds, I think this would be out of scope for this particular issue. Also, since we only have the type `TMetrics` here, creating the right metric would be fairly involved.

Reply:
Thanks. Looks good. 

In the future, we can use the `MetricsStatistics` class to do the averaging so that we're not duplicating functionality. It has the benefit of handling the `PerClassLogLoss`, though I suspect it's not properly handing the class ordering differences.

https://github.com/dotnet/machinelearning/blob/d849ba4c7831586018d5b2cca6bdd2b1fa228ddf/src/Microsoft.ML.Transforms/MetricStatistics.cs#L84-L88